### DISCUSSION TITLE
Navigation Item Authorization

### DISCUSSION THREAD

**Discussion by ousid** [filamentphp/filament#6566] [Help] [answered]
Is there a way to authorize __hide__/__show__ custom navigation item? I searched on the documentation, and in the source code, and I cannot find methods such `visible` or `hidden`like the other class APIs, or maybe I'm missing something?

### Example:

In the `AppServiceProvider`

```php
Filament::serving(function () {
   // ....

   Filament::registerNavigationItems([
     NavigationItem::make('...')
        .....
        // hide, or show this custom navigation item.
  ]);
});

```

Thanks in advance.

**Comment by danharrin** [comment]
You can wrap it in `if ()` ?

**Reply by ousid** [reply]
Hmm, I don't think we can use that

```php
Filament\FilamentManager::Filament\{closure}(): Argument #1 ($item) must be of type Filament\Navigation\NavigationItem, bool given
```

example:

```php
Filament::registerNavigationItems([
    true ?:  NavigationItem::make('...'),
]);
```

**Reply by danharrin** [reply, marked as answer]
```
Filament::registerNavigationItems([
    ...(true ? [NavigationItem::make('...')] : []),
]);
```

or just wrap the whole thing in `if ()`

**Reply by ousid** [reply]
That's may work. Thanks, Dan!

**Reply by ousid** [reply]
@danharrin I have been thinking of contributing in `NavigationItem` by adding something like `visible` and `hidden` methods.

In your opinion, is it something applicable, and/or something worth creating a __PR__ for?

**Reply by danharrin** [reply]
Sure, that sounds fine to me 👍
